modules/integration/tests-common/admin-clients/src/main/java/org/wso2/mb/integration/common/clients/AndesJMSConsumer.java

```diff
@@ -0,0 +1,576 @@
+                    message.acknowledge();
+                    log.info("Acknowledging message : " + message.getJMSMessageID());
+                }
+            }
+
+            if (0 == this.receivedMessageCount.get() % consumerConfig
+                    .getCommitAfterEachMessageCount()) {
+                // Committing session
+                session.commit();
+                log.info("Committed session");
+            } else if (0 == this.receivedMessageCount.get() % consumerConfig
+                    .getRollbackAfterEachMessageCount()) {
+                // Roll-backing session
+                session.rollback();
+                log.info("Roll-backed session");
+            }
+
+            if (this.receivedMessageCount.get() >= consumerConfig
+                    .getUnSubscribeAfterEachMessageCount()) {
+                // Un-Subscribing consumer
+                unSubscribe(true);
+                // Waiting till consumer is un-subscribed so that no messages will be read.
+                AndesClientUtils.sleepForInterval(1000L);
+                return true;
+            } else if (this.receivedMessageCount.get() >= consumerConfig
+                    .getMaximumMessagesToReceived()) {
+                // Stopping the consumer
+                stopClient();
+                // Waiting till consumer is closed so that no messages will be read.
+                AndesClientUtils.sleepForInterval(1000L);
+                return true;
+            }
+
+            // Delaying reading of messages
+            if (0 < consumerConfig.getRunningDelay()) {
+                try {
+                    Thread.sleep(consumerConfig.getRunningDelay());
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Gets the received message count for the consumer.
+     *
+     * @return The received message count.
+     */
+    public long getReceivedMessageCount() {
+        return this.receivedMessageCount.get();
+    }
+
+    /**
+     * Gets the consumer transactions per seconds.
+     *
+     * @return The consumer transactions per seconds.
+     */
+    public double getConsumerTPS() {
+        if (0 == this.lastMessageConsumedTimestamp - this.firstMessageConsumedTimestamp) {
+            return this.receivedMessageCount.doubleValue() / (1D / 1000);
+        } else {
+            return this.receivedMessageCount
+                           .doubleValue() / (((double) (this.lastMessageConsumedTimestamp - this.firstMessageConsumedTimestamp)) / 1000D);
+        }
+    }
+
+    /**
+     * Gets the average latency for the consumer in receiving messages.
+     *
+     * @return The average latency.
+     */
+    public double getAverageLatency() {
+        if (0 == this.receivedMessageCount.doubleValue()) {
+            log.warn("No messages were received");
```

Comment:
There can be many places where the log message 'No messages were received' is valid. Can we change it to something specific like 'No messages were received to calculate average latency'?